cfgov/ask_cfpb/models/search.py

```diff
@@ -7,6 +7,8 @@
     '#', '%', ';', '^', '~', '`', '|',
     '<', '>', '[', ']', '{', '}', '\\'
 ]
+# maximum length of autocomplete form entries before we call a halt
+ASK_AUTOCOMPLETE_MAX = 50
```

Comment:
Is there any way we can set the max length in one place and have it available everywhere we need it? For the front-end, I'm setting the max length in [cfgov/jinja2/v1/_includes/organisms/ask-search.html](https://github.com/cfpb/consumerfinance.gov/pull/6537/files#diff-b908e3e8b32fee11c18f10fa28fc449ad1b3523eafadbbf6a39258e20eb4a3ebR35). Is there one place both the model and template can pull the same max length from?

Reply:
Here's how we do that in find a housing counselor:
- Backend https://github.com/cfpb/consumerfinance.gov/blob/2ace1532d9b24595db6b436f008f02cc7e03ae19/cfgov/housing_counselor/jinja2/housing_counselor/index.html#L395-L403
- Frontend https://github.com/cfpb/consumerfinance.gov/blob/2ace1532d9b24595db6b436f008f02cc7e03ae19/cfgov/unprocessed/apps/find-a-housing-counselor/js/common.js#L6-L8

Other ways to do this might be to:
- Similar to above create a `<script>` tag, but give it an `id` attribute. Include a json string with the data. On the frontend query the page for the script id and then parse the internal json within the tag. (Avoids storing things in the `window`.)
- Creating a JSON API endpoint that the frontend hits via `fetch` to retrieve the settings before it initializes. (Probably the cleanest, but most difficult solution.)

Reply:
@anselmbradford 
Could we expose the limit in jinja2's environment [globals](https://jinja.palletsprojects.com/en/2.11.x/api/#jinja2.Environment.globals), so that at render time any autocomplete box on the site could use our agreed-on character limit?

Reply:
Sure! It would still need to get passed to the JavaScript through one of the means above though.

Reply:
I pushed an implementation of a global limit for back and front ends to this branch.

Reply:
> Sure! It would still need to get passed to the JavaScript through one of the means above though.

I don't so. The [render method](https://github.com/cfpb/consumerfinance.gov/blob/acb41d945c084c3988a99e8892b4cf1b081cec9f/cfgov/jinja2/v1/_includes/organisms/ask-search.html#L35) picks up the global value, which is used to create a [maxlength](https://github.com/cfpb/consumerfinance.gov/blob/acb41d945c084c3988a99e8892b4cf1b081cec9f/cfgov/jinja2/v1/_includes/organisms/ask-search.html#L61) input attribute, which is all the JavaScript needs.

Reply:
Oh sweet! Yea that looks great! Maybe we should do something similar with the mapbox token in HUD 🤔